src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/ICodeGenerationArrayTypeSymbol.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CodeGeneration;
+
+/// <summary>
+/// This interface provides a subset of <see cref="IArrayTypeSymbol"/> for use in code generation scenarios.
+/// </summary>
+internal interface ICodeGenerationArrayTypeSymbol : ICodeGenerationTypeSymbol
```

Comment:
The duplication of all these symbol interfaces is another blocker.  We *created* the interfaces with the consumption/implementation by the IDE as a *primary* use case.  

THe interfaces were *originally* written by IDE team (me) and they moved to compiler with the understanding that these apis were reserved for us to still implement ourselves so we could exactly support these symbolic cases.

Reply:
➡️ This change is part of prerequisite pull request #73675. Resolving this comment as off-topic.